apps/ui/pages/reservation-unit/[id].tsx

```diff
@@ -806,101 +930,94 @@ const ReservationUnit = ({
-  useEffect(() => {
-    if (
-      quickReservationSlot &&
-      initialReservation?.begin &&
-      initialReservation?.end &&
-      (quickReservationSlot.start.toISOString() !== initialReservation.begin ||
-        quickReservationSlot.end.toISOString() !== initialReservation.end)
-    ) {
-      setShouldUnselect((prev) => prev + 1);
-    }
-    // If user resets the calendar/unselects a slot, unselect the quick reservation slot
-    if (quickReservationSlot && !initialReservation) {
-      setShouldUnselect((prev) => prev + 1);
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [initialReservation, setShouldUnselect, setQuickReservationSlot]);
-
   useEffect(() => {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     if ((window as any).cookiehub) {
       window.addEventListener("scroll", onScroll, { passive: true });
     }
-
     return () => {
       window.removeEventListener("scroll", onScroll);
     };
   }, []);
 
-  const futurePricing =
-    reservationUnit != null
-      ? getFuturePricing(reservationUnit, activeApplicationRounds)
-      : undefined;
-  const formatters = getFormatters(i18n.language);
-  const currentDate = focusDate ?? now;
-  const dayStartTime = addHours(startOfDay(currentDate), 6);
-  const equipment = filterNonNullable(reservationUnit?.equipment);
+  useEffect(() => {
+    const scrollToCalendar = () => {
+      if (calendarRef?.current?.parentElement?.offsetTop != null) {
+        window.scroll({
+          top: calendarRef.current.parentElement.offsetTop - 20,
+          behavior: "smooth",
+        });
+      }
+    };
+    if (hash === "calendar" && focusSlot) {
+      scrollToCalendar();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [focusSlot]);
 
   if (reservationUnit == null) {
     return <CenterSpinner />;
   }
-
+  const subventionSuffix = reservationUnit?.canApplyFreeOfCharge ? (
+    <SubventionSuffix
+      placement="reservation-unit-head"
+      setIsDialogOpen={setIsDialogOpen}
+    />
+  ) : undefined;
+  const nextAvailableTime = getNextAvailableTime({
+    day: focusDate,
+    slots: reservableTimeSpans,
+    duration: durationValue,
+    isSlotReservable,
+    reservationUnit,
+  });
+  const LoginAndSubmit = (
+    <LoginFragment
+      isActionDisabled={!focusSlot?.isReservable}
+      apiBaseUrl={apiBaseUrl}
+      actionCallback={() => storeReservationForLogin()}
+      componentIfAuthenticated={
+        <SubmitButton
+          disabled={!focusSlot?.isReservable}
+          type="submit"
+          isLoading={reservationForm.formState.isSubmitting}
+          loadingText={t("reservationCalendar:makeReservationLoading")}
+          data-test="quick-reservation__button--submit"
+        >
+          {t("reservationCalendar:makeReservation")}
+        </SubmitButton>
+      }
+      returnUrl={getPostLoginUrl()}
+    />
+  );
```

Comment:
should not create components inside components

Reply:
That's absolutely correct - extracted it to a function component which is outside the parent component (`submitFragment`).